e2e_correctness/query_neo_mem.py

```diff
@@ -297,3 +301,88 @@ def neo4j_get_graph(neo4j_driver: neo4j.BoltDriver) -> Graph:
     json_data = get_neo4j_data_json(neo4j_driver)
     logger.debug("Building the graph from Neo4j JSON data")
     return create_graph_neo4j_json(json_data)
+
+
+# additions for path testing
+def sort_dict(dict):
+    keys = list(dict.keys())
+    keys.sort()
+    sorted_dict = {i: dict[i] for i in keys}
+    if "id" in sorted_dict:
+        sorted_dict.pop("id")
+    return sorted_dict
+
+
+def execute_query_neo4j(driver, query):
+    with driver.session() as session:
+        query = neo4j.Query(query)
+        results = session.run(query).value()
+    return results
+
+
+def path_to_string_neo4j(path):
+    path_string_list = ["PATH: "]
+
+    n = len(path.nodes)
+
+    for i in range(0, n):
+        node = path.nodes[i]
+        node_labels = list(node.labels)
+        node_labels.sort()
+        node_props = str(sort_dict(node._properties))
+        path_string_list.append("(id:")
+        path_string_list.append(str(node.get("id")) + " labels: ")
+        path_string_list.append(str(node_labels) + " ")
+        path_string_list.append(str(node_props) + ")-")
```

Comment:
`path_string_list.append(f"(id:{str(node.get("id))} "labels: {std(node_labels)} {str(node_props)})-")`
I maybe you don't even need {str(not_string)}, you can just do {not_string} this way but you need to check